lib/rust/ensogl/core/src/data/color/space/def.rs

```diff
@@ -389,6 +389,61 @@ impl Rgb {
         Self::new(r.into() / 255.0, g.into() / 255.0, b.into() / 255.0)
     }
 
+    /// Return a color if the argument is a string matching a `#RGB` or `#RRGGBB` format, where
+    /// `R`, `G`, `B` represent lower- or upper-case hexadecimal digits. The `RR`, `GG`, `BB` color
+    /// components are mapped from `[00 - ff]` value range into `[0.0 - 1.0]` (a three-digit string
+    /// matching a `#RGB` format is equivalent to a six-digit string matching a `#RRGGBB` format,
+    /// constructed by duplicating the digits).
+    ///
+    /// The format is the hexadecimal color notation used in CSS - see:
+    /// https://developer.mozilla.org/en-US/docs/Web/CSS/hex-color
+    /// ```
+    /// # use assert_approx_eq::assert_approx_eq;
+    /// # use ensogl_core::data::color::Rgb;
+    /// # const PRECISION_8_BIT: f32 = 0.001;
+    ///
+    /// let color = Rgb::from_css_hex("#C047AB");
+    /// assert!(color.is_some());
+    /// assert_approx_eq!(color.unwrap().red, 0.753, PRECISION_8_BIT);
+    /// assert_approx_eq!(color.unwrap().green, 0.278, PRECISION_8_BIT);
+    /// assert_approx_eq!(color.unwrap().blue, 0.671, PRECISION_8_BIT);
```

Comment:
IMO these tests are overengineered and hard to read. Instead, take the color, multiply it by 255, round to integer, and check for equality. Colors have 256 values, we don't need to check floats and take into account rounding issues.